packages/peregrine/lib/hooks/usePagination.js

```diff
@@ -21,19 +23,24 @@ const setQueryParam = ({ location, history, parameter, value }) => {
 const defaultInitialPage = 1;
 
 /**
- * `usePagination` provides a pagination state with `currentPage` and
- * `totalPages` as well as an API for interacting with the state.
+ * A [React Hook]{@link https://reactjs.org/docs/hooks-intro.html} that provides
+ * pagination logic.
+ *
+ * Use this hook to implement components that need to navigate through paged
+ * data.
  *
- * @param {Object} location the location object, like window.location, or from react router
- * @param {Object} history the history object, like window.history, or from react router
- * @param {String} namespace the namespace to apply to the pagination query
- * @param {String} parameter the name of the query parameter to use for page
- * @param {Number} initialPage the initial current page value
- * @param {Number} intialTotalPages the total pages expected to be usable by this hook
+ * @kind function
  *
- * TODO update with defaults
+ * @param {Object} config An object containing configuration values
  *
- * @returns {[PaginationState, PaginationApi]}
+ * @param {Object} config.location=window.location The location object, such as window.location or from react router
+ * @param {Object} config.history=window.history The history object, such as window.history or from react router
+ * @param {String} config.namespace='' The namespace to apply to the pagination query
```

Comment:
Can we explain somewhere that if a namespace is provided it will be prepended to the parameter like `?namespace_parameter=`